contracts/scripts/create_genesis.ts

```diff
@@ -82,14 +51,135 @@ const main = () => {
     genesisPath = process.argv[outFlagIndex + 1];
   } else {
     console.log("Setting out genesis path same as base genesis path");
-    genesisPath = baseGenesisPath;
+    genesisPath = path.join(path.dirname(baseGenesisPath), "genesis.json");
   }
 
-  // Address the faucet will be deployed to - address(20)
-  const faucetAddress = "0x0000000000000000000000000000000000000020";
+  return { baseGenesisPath, genesisPath };
+}
 
-  GenesisJson.alloc[faucetAddress.toString()] = createFaucetContractObject();
-  fs.writeFileSync(genesisPath, JSON.stringify(GenesisJson, null, 2));
-};
+async function getArtifact(contractName: string) {
+  const artifactPath = await glob(`**/artifacts/**/${contractName}.json`, {
+    ignore: "node_modules/**",
+  });
+
+  if (artifactPath.length !== 1) {
+    throw Error(`could not find unique artifact for ${contractName}`);
+  }
+
+  return JSON.parse(fs.readFileSync(artifactPath[0], "utf-8"));
+}
+
+// TODO: this is not very efficient,
+// we should be able to narrow down the search once we know
+// where the file will live / how it will be run
+async function getStorageLayout(contractName: string) {
+  // we can read the storage layout from the HH artifacts like this:
+  // 1. read the contracts debug file to get the build-info hash of the most recent compilation
+  // 2. get the full source path of the contract
+  // 3. use the full source path to look up the storage layout in the build-info file
+
+  const debugFile = await glob(`**/artifacts/**/${contractName}.dbg.json`, {
```

Comment:
the path is simply `contracts/artifacts`, but you shouldn't read json but directly access from `HardhatRuntimeEnvironment`